src/main/java/mekanism/common/util/CapabilityUtils.java

```diff
@@ -21,6 +22,38 @@ public static boolean hasCapability(ICapabilityProvider provider, Capability<?>
         return provider.getCapability(cap, side);
     }
 
+    /**
+     * Checks if the capability exists then retrieves it for use
+     *
+     * @param provider  - provider hosting the capability
+     * @param cap       - capability type
+     * @param side      - side to access
+     * @param <T>       - type to return
+     * @return null if not found/valid or cap
+     */
+    public static <T> T get(ICapabilityProvider provider, Capability<T> cap, EnumFacing side) {
+        if (provider == null || cap == null || !hasCapability(provider, cap, side)) {
```

Comment:
`hasCapability` is unnecessary here, `getCapability` just needs to be null checked instead

Reply:
The previous code was running 'has' then 'get'... as well it's proper to run 'has' then 'get' for the API usage.

Reply:
Previous code is legacy on the cards for a rejig. Do you call `containsKey` before you get an element from a `Map`?

From the javadoc: `Basically, this method functions analogously to {@link Map#containsKey(Object)}.`

Reply:
I honestly don't want to argue over it for the 4th or 5th time. We have been over this in DM's on discord before. All I'm doing is replicating existing code, reducing to reusability, and solving the ghosting problem. 

I could have easily just slotted in two ugly block exists calls and been done. However, I decided to spend some extra time by making the code easier to read and more reusable. In which I didn't make assumptions on how you wanted the code. Instead, I replicated what was already in place. Which really is just a 1 line change on your side if you don't like it.

Reply:
Since I want this PR pulled I made the change rather than deal with fighting you over it.